habitat_baselines/train_ppo.py

```diff
@@ -224,137 +225,172 @@ def run_training():
-                ) = actor_critic.act(
-                    step_observation,
-                    rollouts.recurrent_hidden_states[step],
-                    rollouts.masks[step],
-                )
-            pth_time += time() - t_sample_action
+                t_step_env = time()
 
-            t_step_env = time()
+                outputs = envs.step([a[0].item() for a in actions])
+                observations, rewards, dones, infos = [
+                    list(x) for x in zip(*outputs)
+                ]
 
-            outputs = envs.step([a[0].item() for a in actions])
-            observations, rewards, dones, infos = [
-                list(x) for x in zip(*outputs)
-            ]
+                env_time += time() - t_step_env
 
-            env_time += time() - t_step_env
+                t_update_stats = time()
+                batch = batch_obs(observations)
+                rewards = torch.tensor(rewards, dtype=torch.float)
+                rewards = rewards.unsqueeze(1)
 
-            t_update_stats = time()
-            batch = batch_obs(observations)
-            rewards = torch.tensor(rewards, dtype=torch.float)
-            rewards = rewards.unsqueeze(1)
+                masks = torch.tensor(
+                    [[0.0] if done else [1.0] for done in dones],
+                    dtype=torch.float,
+                )
 
-            masks = torch.tensor(
-                [[0.0] if done else [1.0] for done in dones], dtype=torch.float
-            )
+                current_episode_reward += rewards
+                episode_rewards += (1 - masks) * current_episode_reward
+                episode_counts += 1 - masks
+                current_episode_reward *= masks
 
-            current_episode_reward += rewards
-            episode_rewards += (1 - masks) * current_episode_reward
-            episode_counts += 1 - masks
-            current_episode_reward *= masks
-
-            rollouts.insert(
-                batch,
-                recurrent_hidden_states,
-                actions,
-                actions_log_probs,
-                values,
-                rewards,
-                masks,
-            )
+                rollouts.insert(
+                    batch,
+                    recurrent_hidden_states,
+                    actions,
+                    actions_log_probs,
+                    values,
+                    rewards,
+                    masks,
+                )
 
-            count_steps += envs.num_envs
-            pth_time += time() - t_update_stats
+                count_steps += envs.num_envs
+                pth_time += time() - t_update_stats
 
-        if len(window_episode_reward) == args.reward_window_size:
-            window_episode_reward.popleft()
-            window_episode_counts.popleft()
-        window_episode_reward.append(episode_rewards.clone())
-        window_episode_counts.append(episode_counts.clone())
+            if len(window_episode_reward) == args.reward_window_size:
```

Comment:
You can get rid of this by adding `maxlen=args.reward_window_size` to the constructor of the `deque`